app/assets/javascripts/views/searches/results/results_cross.js

```diff
@@ -206,16 +221,32 @@
 
       var popups = preparePopup(resultsJson);
 
-      function styler(entry){
-        return counter < 20 ? {
-          markerColor: colors[counter++],
-          iconColor: 'white',
-          icon: 'info',
-          text: popups[entry.id]
-        } : null;
-      }
-      return styler;
+      var colors_by_row = {};
+
+      //modified section
+      function styler(entry) {
+        if(colors_by_row[rows_by_ling[entry.id]]) {
```

Comment:
Can we refactor a little bit here?

Reply:
Something like:

Reply:
This looks good to me.  I've added this to a second commit.